### ISSUE TITLE
Acquire a more current Windows machine

### ISSUE THREAD

**Issue by WillClinger** [larcenists/larceny#701]
Changeset 981f4d70c3703446939d3af28c6b30e0aae66b6d made several changes needed to build Larceny on our current Windows XP machine, an IBM (!) ThinkPad A31p circa 2002.  Some of those changes corrected portability problems such as declaring a C identifier somewhere other than at the head of a block.

Some other changes limited a change made by Felix or Lars to the ARM port.  Several of those now ARM-specific changes should be active everywhere except on Windows, but I couldn't figure out how to restrict them to non-Windows builds.  (`#ifndef WIN32` didn't seem to work.)  Those patches are temporary and should be reconsidered soon.

Another temporary patch disabled the `list-directory` code in "src/Rts/Sys/osdep-win32.c".  That code needs to be replaced by Windows-specific code I didn't have time to write last night.

Most importantly, I had to disable the DEP-related code in "src/Rts/Sys/larceny.c" because I couldn't find the right library on our antiquated Windows machine.  I know that code works because I built it and tested it on a borrowed Windows machine before adding it to Larceny's code base.

We need to beg, borrow, or otherwise acquire a more current Windows machine.

**Comment by lars-t-hansen** [comment]
I have a newish machine with Windows 7.  But I probably also have only recent C compilers (gcc 4.8 under Cygwin and probably Visual Studio 2012, at a guess) so I might not find all the problems that way.

Even Microsoft C has allowed names to be declared within blocks for a while, unless I'm mistaken.  What compiler/toolchain/environment are you using?

(Is Windows XP really a target, or just what you use because it's available to you?)

**Comment by WillClinger** [comment]
In a few days, I'll have a Windows 7 laptop on long-term loan.  I filed this issue mainly so I wouldn't forget what needed to be undone from the changes I pushed last night.

On the IBM ThinkPad A31p, I'm running Microsoft Platform SDK Windows Server 2003 32-bit IE 6.0 or Windows XP 32-bit IE 5.5.  They behave the same so far as I can tell.  I ran into several problems with unsupported headers and such (e.g. `<dirent.h>`, `<stdint.h>`, and Gnu-specific annotations), not just the declarations-not-allowed-in-the-middle-of-blocks problem.  (I think I've run into the declarations problem on other platforms as well, possibly MacOS X PowerPC, but I don't remember the details.)

I'm using the Windows XP machine because it's the machine I've been using for Windows builds these past 12 years, and I'm too cheap to pay actual money for a Windows machine.  It will be nice if v0.98 can run under Windows XP, but the real problem is building v0.98 so it can run under Vista and later.
